src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx

```diff
@@ -764,18 +777,54 @@ export class SavedObjectsTable extends Component<SavedObjectsTableProps, SavedOb
       isSearching,
       savedObjectCounts,
     } = this.state;
-    const { http, allowedTypes, applications } = this.props;
+    const { http, allowedTypes, applications, namespaceRegistry } = this.props;
 
     const selectionConfig = {
       onSelectionChange: this.onSelectionChanged,
     };
+    const typeCounts = savedObjectCounts.type || {};
 
     const filterOptions = allowedTypes.map((type) => ({
       value: type,
       name: type,
-      view: `${type} (${savedObjectCounts[type] || 0})`,
+      view: `${type} (${typeCounts[type] || 0})`,
     }));
 
+    const filters = [
+      {
+        type: 'field_value_selection',
+        field: 'type',
+        name: i18n.translate('savedObjectsManagement.objectsTable.table.typeFilterName', {
+          defaultMessage: 'Type',
+        }),
+        multiSelect: 'or',
+        options: filterOptions,
+      },
+    ];
+
+    const availableNamespaces = namespaceRegistry.getAll() || [];
+    if (availableNamespaces && availableNamespaces.length > 0) {
+      const nsCounts = savedObjectCounts.namespaces || {};
+      const nsFilterOptions = availableNamespaces.map((ns) => {
+        return {
+          name: ns.name,
+          value: ns.id,
+          view: `${ns.name} (${nsCounts[ns.id] || 0})`,
+        };
+      });
+
+      const filterText = namespaceRegistry.getAlias() || 'Namespaces';
+      filters.push({
+        type: 'field_value_selection',
+        field: 'namespaces',
+        name: i18n.translate('savedObjectsManagement.objectsTable.table.namespaceFilterName', {
+          defaultMessage: filterText,
+        }),
```

Comment:
So by registering an alias, you are only able to override the default message here. Any translation file, including english will override it. Not sure what the best course of action here is. Do we want it to always override? In that case how would we handle translations.

Reply:
I added a check to see if an alias has been registered and throw an error now. This is what you see when trying to run OSD if an alias has been registered more than once.

![Screen Shot 2022-10-27 at 4 23 33 PM](https://user-images.githubusercontent.com/17432265/198391043-b0f0c9d3-67ed-483e-a5ee-43b20ce3e65d.png)

Reply:
I'm okay with that. If thats the approach, we should move the `i18n.translate` to where we default to "Namespace", and just set the value here

Reply:
Updated to be `namespaceRegistry.getAlias() || i18n.translate...`